src/solver/conv_winoRxS_ultra_f2x3.cpp

```diff
@@ -0,0 +1,653 @@
+
+    return works * group_size * sizeof(uint32_t);
+}
+
+ConvSolution ConvBinWinogradUltraRxSf2x3::GetSolution(const ExecutionContext& ctx,
+                                                      const ProblemDescription& problem) const
+{
+    const unsigned n_groups = ctx.GetStream().GetMaxHardwareComputeUnits();
+    const auto desc         = WinogradUltraDescription(problem);
+    const auto intl_factor  = 1;
+
+    uint64_t reserved_offset = 0;
+    int* reserved_ptr        = nullptr;
+    float relu_alpha         = 1.0;
+
+    std::vector<uint32_t> control_buf;
+    WU_control_make_3x3(desc, control_buf, n_groups, intl_factor);
+
+    const unsigned n_works     = control_buf.size() / group_size;
+    const size_t workspace_req = GetWorkspaceSize(problem);
+    const size_t workspace_buf = control_buf.size() * sizeof(decltype(control_buf)::value_type);
+
+    if(workspace_req != workspace_buf)
+        MIOPEN_THROW("ConvBinWinogradUltraRxSf2x3 invalid control buffer size");
+
+    const size_t wg_size = 256;
+
+    KernelInfo kernel;
+
+    kernel.g_wk.push_back(wg_size * n_groups * problem.group_counts);
+    kernel.g_wk.push_back(1);
+    kernel.g_wk.push_back(1);
+
+    kernel.l_wk.push_back(wg_size);
+    kernel.l_wk.push_back(1);
+    kernel.l_wk.push_back(1);
+
+    const std::string kernel_name    = "miopenSp3AsmConv_Ultra_v1_0_14_gfx10";
+    const std::string kernel_file    = "Conv_Winograd_Ultra_v1_0_14";
+    const std::string kernel_postfix = "_fp16_pk_stride1";
+
+    kernel.kernel_name = kernel_name + kernel_postfix;
+    kernel.kernel_file = kernel_file + kernel_postfix + ".s";
+
+    KernelBuildParameters options{
+        {"ROCM_METADATA_VERSION", 5},
+        {"control_buf_alloc_type", 0},
+        {kbp::Option{}, "mcumode"},
+        {kbp::Option{}, "mwavefrontsize64"},
+    };
+    kernel.comp_options = options.GenerateFor(kbp::GcnAsm{});
+
+    ConvSolution solution;
+
+    solution.workspace_sz = workspace_req;
+    solution.construction_params.push_back(kernel);
+
+    solution.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_params) {
+            const auto kern           = handle.Run(kernels.front());
+            const auto& invoke_params = primitive_params.CastTo<conv::DataInvokeParams>();
+            const auto& tensors       = invoke_params.tensors;
+
+            const auto workspace_ptr  = invoke_params.workSpace;
+            const auto workspace_size = invoke_params.workSpaceSize;
+
+            if((workspace_ptr == nullptr && workspace_req > 0) || workspace_size < workspace_req)
+                MIOPEN_THROW("Not enough workspace for Winograd Ultra (" +
+                             std::to_string(workspace_size) + " provided, " +
+                             std::to_string(workspace_req) + " required)");
+
+            CopyDataToBuffer(ctx.GetStream(), control_buf, workspace_ptr);

```

Comment:
```suggestion
            CopyDataToBuffer(handle, control_buf, workspace_ptr);
```

Reply:
Thank you, fixed in [62d7d10](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1510/commits/62d7d10b9bfd8e58b43e4fcda5248eaa56722a63)!